src/buffer.h

```diff
@@ -66,27 +66,31 @@ class Buffer : public std::enable_shared_from_this<Buffer> {
 };
 
 
-template<bool CLEAN_AT_END>
-class MemoryBuffer : public Buffer {
+class MemoryRegionBuffer : public Buffer {
```

Comment:
I don't find `MemoryRegionBuffer` name explicit.
Maybe more explicit names like `SharedRegionBuffer` and `OwnedRegionBuffer` ?

Reply:
I had considered `SharedRegionBuffer`, but for me in the context of C++ the term *shared* implies shared ownership, whereas `MemoryRegionBuffer` is just a buffer referencing an existing memory region without any claim of ownership.

Reply:
Maybe `MemoryViewBuffer` (or simply `MemoryView`) instead of `SharedRegionBuffer` ?

Reply:
That's a matter of taste. I like my version better.

Reply:
While the C++ type is indeed a `Buffer`, this is not really a buffer. This is a view on a buffer.
I don't see how `MemoryRegionBuffer` imply that the buffer doesn't how the memory region.

The view name is explicit. (And this is the terminology used elsewhere, [memoryview in python](https://docs.python.org/3.8/library/stdtypes.html#typememoryview) and [std::string_view](https://en.cppreference.com/w/cpp/string/basic_string_view))

Reply:
Renamed `MemoryRegionBuffer` to `MemoryViewBuffer`, and `AllocatedMemoryBuffer` to `MemoryBuffer` in two separate commits.